### PR TITLE
Special case for Unallocated aliased fields when filtering

### PR DESCRIPTION
**Pull request by ghost**
Added special case for unallocated/unassigned aliased fields when filtering.

## What does this PR change?
* This PR modifies the ops returned by convertAliasFilterToLabelAnnotationFilter, to handle the special case where a user wants to examine unallocated/unassigned workloads within the aliased aggregation category.

## Does this PR relate to any other PRs?
* Nope

## How will this PR impact users?
* If it was actually a regression, fixes the allocation view to so you can view unallocated workloads. If it wasn't a regression, added support for doing that with v2.1 filters to the backend.

## Does this PR address any GitHub or Zendesk issues?
* Closes [GTM-38](https://kubecost.atlassian.net/browse/GTM-38)

## How was this PR tested?
* Manually, against example failing queries used to replicate the original issue.

## Does this PR require changes to documentation?
* I don't think so

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
* No, because it won't let me add labels.

### CODE REVIEW

**Review by michaelmdresser — COMMENTED** [review]
Interesting fix! Is there a way to write a unit test for this? I know there are filter tests scattered around, I'm sorry I can't recall exactly the right place to look.

File: pkg/kubecost/allocationmatcher.go [hunk 1]
```diff
@@ -256,5 +258,17 @@ func convertAliasFilterToLabelAnnotationFilter(aliasKey string, filterValue stri
 				annotationOp,
 			),
 		),
-	), nil
+	)
+	var node ast.FilterNode
+	if filterValue == "" || filterValue == "__unallocated__" {
+		node = ops.Or(
```
Comment (nikovacevic) [inline pkg/kubecost/allocationmatcher.go:272]: Can we add a brief comment here about the non-extant case? Something that describes why we need these specific filter nodes for this specific condition? (It's described well in the PR description, but future authors may appreciate a brief comment in the code.)

**Review by nikovacevic — APPROVED** [review]
As we've discussed, this should work! Only ask is a small bit more commentary.

File: pkg/kubecost/allocationmatcher.go (hunk 1 above)
Comment (ghost) [reply pkg/kubecost/allocationmatcher.go:272]: Added, I'd meant to do that, woops.

Comment (nikovacevic) [reply pkg/kubecost/allocationmatcher.go:272]: Excellent, looks great!

**Pull request merged**